app/views/companies/_form.html.erb

```diff
@@ -62,6 +62,11 @@
       <%= f.file_field :logo %>
       <%= f.hidden_field :logo_cache %>
     </div>
+    <%= f.label :intro_image, "Company Image:", class: "col-md-2 control-label" %>
+    <div class="col-md-4">    
+      <%= f.file_field :intro_image %>
+      <%= f.hidden_field :intro_image_cache %>
```

Comment:
@arafatm in case any of the form validation fails normally the image uploaded tend to dis appear so `image_cache` will keep track of that by re-storing the image user uploaded prior to failed validation that way he doesn't have to upload again after fixing the failed validation field